addons/mail/models/mail_compose_message.py

```diff
@@ -998,15 +1137,20 @@ def _prepare_mail_values_dynamic(self, res_ids):
 
         # generate template-based values
         if self.template_id:
+            fields_to_render = [
+                'attachment_ids',
+                'email_to',
+                'email_cc',
+                'partner_ids',
+                'report_template_ids',
+            ]
+            # don't schedule mail again if composer was scheduled
+            if self.state != 'scheduled':
+                fields_to_render.append('scheduled_date')
```

Comment:
That is something to discuss (continuing scratching my head). I may schedule a composer for tomorrow, with a template asking to schedule one day ahead. What is the final sending time, one day or two days ahead ? You have 2 hours :D .

I would say 2 days. One schedule day to execute the composer, then one schedule day linked to template definition (hence two fields options). But that is open to discussion.